packages/backend/src/modules/tvl/ChainTvlSubmodule.ts

```diff
@@ -0,0 +1,114 @@
+import { HttpClient, Logger, UniversalEtherscanClient } from '@l2beat/shared'
+import { ChainId, ProjectId, Token } from '@l2beat/shared-pure'
+import { providers } from 'ethers'
+
+import { ChainTvlConfig } from '../../config/Config'
+import { TotalSupplyFormulaUpdater } from '../../core/assets/TotalSupplyFormulaUpdater'
+import { BlockNumberUpdater } from '../../core/BlockNumberUpdater'
+import { Clock } from '../../core/Clock'
+import { PriceUpdater } from '../../core/PriceUpdater'
+import { TotalSupplyProvider } from '../../core/totalSupply/TotalSupplyProvider'
+import { TotalSupplyUpdater } from '../../core/totalSupply/TotalSupplyUpdater'
+import { EthereumClient } from '../../peripherals/ethereum/EthereumClient'
+import { MulticallClient } from '../../peripherals/ethereum/multicall/MulticallClient'
+import { MulticallConfigEntry } from '../../peripherals/ethereum/multicall/types'
+import { TvlSubmodule } from '../ApplicationModule'
+import { TvlDatabase } from './types'
+
+export function chainTvlSubmodule(
+  chainId: ChainId,
+  projectId: ProjectId,
+  chainTvlConfig: ChainTvlConfig | false,
+  tokens: Token[],
+  multicallConfig: MulticallConfigEntry[],
+  db: TvlDatabase,
+  priceUpdater: PriceUpdater,
+  http: HttpClient,
+  clock: Clock,
+  logger: Logger,
+): TvlSubmodule | undefined {
+  const name = `${ChainId.getName(chainId)}TvlModule`
```

Comment:
Capitalize the first letter and extract this logic to a function

Reply:
[4b4efe5](https://github.com/l2beat/l2beat/pull/1889/commits/4b4efe527e7be0a05cd9eb5fcdcfe49e2ce59dae)